finder/tagged.go

```diff
@@ -226,14 +262,54 @@ func ParseTaggedConditions(conditions []string) ([]TaggedTerm, error) {
 		default:
 			return nil, fmt.Errorf("wrong seriesByTag expr: %#v", s)
 		}
+		if len(taggedCosts) > 0 {
+			if costs, ok := taggedCosts[terms[i].Key]; ok {
+				setCost(&terms[i], costs)
+			}
+		}
 	}
 
-	sort.Sort(TaggedTermList(terms))
+	if len(taggedCosts) == 0 {
+		sort.Sort(TaggedTermList(terms))
+	} else {
+		// compare with taggs costs
+		sort.Slice(terms, func(i, j int) bool {
+			eq, comparable := lessCosts(terms, i, j)
+			if comparable {
+				return eq
+			}
+
+			if terms[i].Op < terms[j].Op {
+				return true
+			}
+			if terms[i].Op > terms[j].Op {
```

Comment:
Also, overall it seems that closes should be moved to the bottom of this function as it seems to be that line 305 shouldn't be reachable at all.

Reply:
May be cleanup block and use `if terms[i].Op == terms[j].Op` for select code block ?